compiler/rustc_data_structures/src/sip128.rs

```diff
@@ -412,52 +409,52 @@ impl SipHasher128 {
 impl Hasher for SipHasher128 {
     #[inline]
     fn write_u8(&mut self, i: u8) {
-        self.short_write(i);
```

Comment:
Moving the endianness conversion from `StableHasher` to `SipHasher128` probably means some documentation needs to be updated. Currently it's `StableHasher` that promises to take care of platform issues.

This doesn't seem to be a big issue though since StableHasher is the only user of the internal SipHasher128. But it's public, so some other rustc crate could start using it in the future.

Reply:
I guess that we have two options. Either we can make `short_write` public and uphold the endianness guarantees in `StableHasher`, or keep it like I already have it in this PR, but change the documentation.

Any preference?

Reply:
Making short_write public seems better since then the platform-independence concerns are all handled in `StableHasher`

Reply:
I agree, I changed it to use `to_le_bytes` in `StableHasher` and `to_ne_bytes` in `SipHasher128`.